scipy/io/tests/test_mmio.py

```diff
@@ -28,6 +28,22 @@ def test_simple(self):
         b = mmread(fn)
         assert_array_almost_equal(a,b)
 
+    def test_simple_upper_triangle(self):
+        a = [[0,1],[0,0]]
```

Comment:
Maybe use `np.triu, np.tril` to generate upper and lower triangular matrices of different sizes, and loop over these testing round trip?  Convert to list and sparse arrays to check these are also correctly written / read?